youtube_dl/extractor/beampro.py

```diff
@@ -6,15 +6,17 @@
     ExtractorError,
     clean_html,
     compat_str,
+    float_or_none,
     int_or_none,
     parse_iso8601,
     try_get,
+    urljoin,
 )
 
 
 class BeamProLiveIE(InfoExtractor):
     IE_NAME = 'Beam:live'
-    _VALID_URL = r'https?://(?:\w+\.)?beam\.pro/(?P<id>[^/?#&]+)'
+    _VALID_URL = r'https?://(?:\w+\.)?beam\.pro/(?P<id>[^/?#&]+)$'
```

Comment:
Should match `http://www.beam.pro/niterhayven/` and `http://www.beam.pro/niterhayven?abc=123`.